### PR TITLE
update nlcglib package

### PR DESCRIPTION
**Pull request by simonpintarelli**
- update package for `nlcglib`
- propagate `+rocm`, `+cuda` from sirius to nlcglib

### CODE REVIEW

**Review by toxa81 — DISMISSED** [review]
LGTM!

File: var/spack/repos/builtin/packages/nlcglib/package.py [hunk 1]
```diff
@@ -1,66 +1,102 @@
-# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (haampie) [inline var/spack/repos/builtin/packages/nlcglib/package.py]:
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

**Review by haampie — APPROVED** [review]

**Comment by simonpintarelli** [comment]
@haampie Thanks for the review. Can we merge it? Or do we need a second approval of @toxa81?

**Review by toxa81 — APPROVED** [review]
LGTM

**Comment by simonpintarelli** [comment]
Ping @haampie :)

**Pull request merged**